README.md

```diff
@@ -129,37 +200,26 @@ Read more about [sbt release](https://github.com/sbt/sbt-release)
 
-2. Disable automatic compaction for selected ACID tables: To do this, you can set a table property using the ALTER TABLE command:
+2. This snapshot of list of files is created at the RDD level. These snapshot are at the RDD level so even when using same table in single SQL it may be operating on two different snapshots
 
-        ALTER TABLE <> SET TBLPROPERTIES ("NO_AUTO_COMPACTION"="true")  
+    spark.sql("select * from a join a)
 
-This will disable automatic compaction on a particular table, and you can use this approach if you have a limited set of ACID tables that you intend to access using this datasource.
+3. The files in the snapshot needs to be protected till the RDD is in use. By design concurrent reads and writes on the Hive ACID works with the help of locks, where every client (across multiple engines) that is operating on ACID tables is expected to acquire locks for the duration of reads and writes. The lifetime of RDD can be very long, to avoid blocking other operations like inserts this datasource _DOES NOT_ acquire lock but uses an alternative mechanism to protect reads. Other way the snapshot can be protected is by making sure the files in the snapshot are not deleted while in use. For the current datasoure any table on which Spark is operating `Automatic Cleanup` should be disabled. This makes sure that `compaction` is performed but cleaner does not clean any file. To disable automatic cleaner on table
 
-Once you have disabled automatic compaction either globally or on a particular set of tables, you can chose to run compaction manually at a desired time when you know there are no readers reading these acid tables, using an ALTER TABLE command:
+         ALTER TABLE <> SET TBLPROPERTIES ("NO_CLEANER"="true")
 
-        ALTER TABLE table_name [PARTITION (partition_key = 'partition_value' [, ...])] COMPACT 'compaction_type'[AND WAIT] [WITH OVERWRITE TBLPROPERTIES ("property"="value" [, ...])];
+	When the table is not in use cleaner can be enabled and all the files that needs cleaned will get queued up for cleaner. Disabling cleaner has no performance implication on reads/writes, only the storage requirement will be higher as system may hold on to all the snapshots in use.
 
-compaction_type are either `MAJOR` or `MINOR`
 
-More details on the above commands and their variations available [here](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL).
+4. Note that even though reads are protected admin operation like `TRUNCATE` `ALTER TABLE DROP COLUMN` and `DROP` have no protected as they clean files with intevention from cleaner. These operations should be performed when Spark is not using the table.
```

Comment:
change from `have no protected` to  `have no protection`

Reply:
fixed